### ISSUE TITLE
Difficulty properly reading in input file, and subsequent bug

### ISSUE THREAD

**Issue by prahaladvenkateshan** [Turox/VRPTW_CPP#1]
Hi Adriel,

I cloned the repo and in stepping through the code as is, and setting up a breakpoint on 

https://github.com/ADCDS/VRPTW_CPP/blob/b91534867d77415e9908524335a5ef9a35880bbb/VRPTW/Utils.cpp#L51

I observe that as expected, `Utils::raw_rows.size() = 1001` for a 1000 customer problem. 

Then, I tried to run this for RC208-25 node problem, with input file as attached 
[rc208.txt](https://github.com/ADCDS/VRPTW_CPP/files/14906293/rc208.txt)

even though the file has 25 customers, `Utils::raw_rows.size() = 27`, while I would expect it to be size 26.

On checking this further, it appears that the 25th customer's data is being repeated in `raw_rows[25]` and `raw_rows[26]`. 

Then, I changed the code so that for this input file, `raw_rows[26]` is not populated, just to see how the code performs on the problem. I did the following change:

```
	if (Utils::raw_rows.size() < 26)
		Utils::raw_rows.push_back(new Point(aux[0], aux[1], aux[2], aux[3], aux[4], aux[5], aux[6]));
```

Then, there is a segfault at the following line:

https://github.com/ADCDS/VRPTW_CPP/blob/b91534867d77415e9908524335a5ef9a35880bbb/VRPTW/Solution.cpp#L136

with the error being:

`Unhandled exception at 0x00007FF6DCA73FD1 in VRPTW.exe: 0xC0000094: Integer division by zero.`

The call stack when this error occurs is :

```
	VRPTW.exe!Solution::mutate_swap() Line 136	C++
 	VRPTW.exe!Solution::mutate() Line 254	C++
 	VRPTW.exe!Genetic::run(int max_iter) Line 272	C++
 	VRPTW.exe!main::__l4::<lambda>() Line 53	C++
 	[External Code]	

```

Could you please help diagnose this problem?

Thanks for your time.
Prahalad

**Comment by ADCDS** [comment]
Hello @prahaladvenkateshan,
Thanks for opening this issue, this project is a really old code for college assignments.
I spent the entire afternoon trying to fix the issues. I've came to the conclusion that this code is really really bad :D, i dont even know how the teacher approved this code.

Well, I did the following things: 
1- Converted the project to Cmake
2 - Fixed the ```rand() % (vehicles.size() - 2 + 1) + 1;``` error: This line of code belongs to a function that mutates a current solution for the VRPTW context. It should select a random point (customer) remove it from its original position and try to insert into another vehicle's route. The line now is ```(rand() % (Utils::raw_rows.size() - 1)) + 1```, it cannot select the depot (customer with id == 0) hence the ```- 1``` and then ```+1```
3 - Thats the sad part, I was not able to fix the code and I dont have anymore time to spend trying to fix it :/ I've left a series of comments inside the code that may help debugging:

https://github.com/ADCDS/VRPTW_CPP/blob/2902163ce22b545603938439c58997602faf7515/Solution.cpp#L59

https://github.com/ADCDS/VRPTW_CPP/blob/2902163ce22b545603938439c58997602faf7515/Vehicle.cpp#L64
